packages/react-sdk/docusaurus/docs/React/02-tutorials/02-audio-room.mdx

```diff
@@ -3,4 +3,543 @@ title: Audio Room Tutorial
+:::
+
+Next, we want to implement a button to allow the user to mute and un-mute. This requires a few things. We need to check if they are currently sharing their audio and change the button accordingly.
+
+Here's an example how to do this:
+
+```tsx
+const currentUser = participants.find((p) => p.userId === user?.id);
+const hasAudio = currentUser?.publishedTracks.includes(
+  SfuModels.TrackType.AUDIO,
+);
+
+/* more code */
+<button onClick={() => muteUser()}>{hasAudio ? 'Mute' : 'Unmute'}</button>;
+```
+
+:::note
+There are different techniques to determine the current user. We are accessing the `userId` through a `UserContext` provider object, but feel free to take different approaches.
+:::
+
+Now, the `muteUser()` function also has to check if the user is currently sharing audio.
+If yes, we can use the `stopPublishingAudio` function from the `useMediaDevices` hook.
+Otherwise, we can get the current audio stream of the user through `getAudioStream()` and then publish it.
+
+Here's the code to do it:
+
+```tsx
+async function muteUser() {
+  if (hasAudio) {
+    stopPublishingAudio();
+  } else {
+    const audioStream = await getAudioStream();
+    await call?.publishAudioStream(audioStream);
+  }
+}
+```
+
+The last thing we want to do is to list know when the call ends. We can use `call.on` to listen to certain events.
+In our case, we want to listen to the `call.ended` event.
+
+:::tip
+You can find all other event types on the [Socket Events](../../guides/04-socket-events) page.
+:::
+
+Here's how to subscribe to it in a `useEffect` (and unsubscribe when the component gets un-mounted):
+
+```tsx
+useEffect(() => {
+  const unsubscribe = call?.on('call.ended', (event: StreamCallEvent) => {
+    // replace with your custom leave function
+    leave();
+  });
+
+  return () => {
+    if (unsubscribe) {
+      unsubscribe();
+    }
+  };
+}, [call, leave]);
+```
+
+With that, we've built up the UI for the audio room itself. The last thing is to handle speaking requests from
+both participants' and hosts' sides.
+
+### Managing speaking requests
+
+There are two roles here. The participants want to be able to request permission to speak. We need to identify if a participant can send audio and can request permission.
+We can do it like this:
+
+```tsx
+const canSendAudio = useHasPermissions(OwnCapability.SEND_AUDIO);
+const canRequestSpeakingPermissions = call?.permissionsContext.canRequest(
+  OwnCapability.SEND_AUDIO,
+);
+```
+
+Next, the code to request permission is a simple button and a call to `requestPermissions` on the `call` object:
+
+```tsx
+<button onClick={() => requestSpeakingPermission()}>Raise hand</button>;
+
+/* more code */
+async function requestSpeakingPermission() {
+  await call?.requestPermissions({
+    permissions: [OwnCapability.SEND_AUDIO],
+  });
+}
+```
+
+Same as in the previous chapter, we can listen to new events of type `call.permission_request` to get notified when there's a new request coming in.
```

Comment:
This is new, but we have the `usePermissionRequest` hook to avoid subscribing to WS events